src/packages/api/api_manager.py

```diff
@@ -0,0 +1,468 @@
+"""
+api_manager.py - Provide functions for communicating with the Fuel Rats API.
+
+Copyright (c) 2018 The Fuel Rat Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+"""
+
+import asyncio
+from datetime import datetime
+from html import escape
+import logging
+import json
+import typing
+from uuid import UUID
+
+import aiohttp
+
+from src.config import CONFIG_MARKER
+from src.packages.mark_for_deletion import MarkForDeletion
+from src.packages.rat import Rat
+from src.packages.quotation import Quotation
+from src.packages.rescue import Rescue
+from src.packages.utils import Platforms, Status
+
+LOG = logging.getLogger(f"mecha.{__name__}")
+
+
+class APIManager:
+    """
+    Worker class to interface with the Fuel Rats API.
+    """
+    _config: typing.ClassVar[typing.Dict]
+    _base_url: typing.ClassVar[str]
+    _online_mode: typing.ClassVar[bool]
+    MAX_RETRIES: typing.ClassVar[int] = 3
+    TIMEOUT: typing.ClassVar[aiohttp.ClientTimeout] = aiohttp.ClientTimeout(total=10)
+
+    @classmethod
+    @CONFIG_MARKER
+    def rehash_handler(cls, data: typing.Dict):
+        """
+        Apply new configuration data
+
+        Args:
+            data (Dict): New configuration data
+        """
+        cls._config = data
+        cls._base_url = data['api']['url']
+        cls._online_mode = data['api']['online_mode']
+        if not cls._base_url.endswith('/'):
+            cls._base_url = f"{cls._base_url}/"
+
+    @classmethod
+    @CONFIG_MARKER
+    def validate_config(cls, data: typing.Dict):
+        """
+        Validate the API's configuration values.
+
+        Args:
+            data (Dict): Configuration options
+        """
+        if 'api' not in data.keys():
+            raise ValueError("[API] Config section 'api' could not be found!")
+
+        api_data = data['api']
+
+        if not isinstance(api_data['url'], str):
+            raise ValueError("[API] API 'url' value must be a string.")
+
+        if not isinstance(api_data['online_mode'], bool):
+            raise ValueError("[API] API 'online_mode' must be true or false.")
+
+        if not api_data['url'].startswith('http'):
```

Comment:
shouldn't we be enforcing `HTTPS://` ?

Reply:
I honestly think that's not for Mecha to decide. Using plaintext HTTP could be completely valid if the API was hosted in a local network.

Reply:
FWIW The website internally communicates over HTTP with the API since they're both on the same host. Our API proxy system communicates with the client over HTTPS, which relays the request to the API over HTTP.

Reply:
I still don't think its a good idea to use http where https can be used. of course, since all this is probably going to end up running on the same set of internal servers i suppose it makes little difference.
🤷‍♂

Reply:
The point that we're trying to make here is that the client should not enforce the scheme. Yes, it makes perfect sense to only use HTTPS. We're only going to use HTTPS. But the software should not lock it down to that through hardcode. At most, it should fire warnings or require a special config directive to allow unsecure HTTP.